common-kafka/src/test/java/com/cerner/common/kafka/consumer/ConsumerOffsetClientTest.java

```diff
@@ -230,6 +230,41 @@ public void getOffsetsForTimes() {
         requestValue.values().forEach(i -> assertThat(i, is(time)));
     }
 
+    @Test
+    public void getOffsetsForTimesNullOffsets() {
+        Map<TopicPartition, OffsetAndTimestamp> offsets = new HashMap<>();
+        offsets.put(new TopicPartition("topic1", 0), null);
+        offsets.put(new TopicPartition("topic1", 1), new OffsetAndTimestamp(234L, 10));
+
+
+        when(consumer.partitionsFor("topic1")).thenReturn(Arrays.asList(
+                new PartitionInfo("topic1", 0, null, null, null),
+                new PartitionInfo("topic1", 1, null, null, null)));
+
+        when(consumer.offsetsForTimes(anyObject())).thenReturn(offsets);
+
+        Map<TopicPartition, Long> longOffsets = new HashMap<>();
+        longOffsets.put(new TopicPartition("topic1", 1), 234L);
```

Comment:
Shouldn't this have called `consumer.endOffsets(..)` for partition 0? Thats what I thought the last few lines of our method did

Reply:
Right, line 265 shows that the endOffsets has topic1, partition 0. Since the foundOffsets is null it is defaulting the offsetForTime to endOffsets.

Reply:
Shouldn't we then have a value in `longOffsets` map here with partition 0 and the end offset value?